src/app/cube/core/route-backwards-compat.guard.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, UrlTree, Router } from '@angular/router';
+import { Observable } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { environment } from '@env/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RouteBackwardsCompatGuard implements CanActivate {
+
+  constructor(private router: Router, private http: HttpClient) {}
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+      const cuidRegex = /([a-z0-9]){8}-([a-z0-9]){4}-([a-z0-9]){4}-([a-z0-9]){4}-([a-z0-9]){12}/;
+      if (next.params.learningObjectName.search(cuidRegex) === 0) {
+        return true;
+      } else {
+        return this.http.get(`${environment.apiURL}}/learning-objects/${next.params.username}/${next.params.learningObjectName}`, { withCredentials: false }).toPromise().then(cuid => {
+          return this.router.createUrlTree([`/details/${next.params.username}/${cuid}`]);
+        }).catch(err => {
```

Comment:
It looks like we are catching and swallowing the error here.
I would like to see an error translation so that we can clearly communicate the error to the user.

Reply:
Added the toaster service to the guard, now if you try to route to something that doesn't make sense like /details/username/demodemo, then you will be rerouted to home with a corresponding error message from the toaster.